### PR TITLE
Allow slicing notation in audio file names

### PR DESCRIPTION
**Pull request by thequilo**
Add notation to select channels and a slice from the audio path passed to `load_audio`

### CODE REVIEW

File: paderbox/io/audioread.py [hunk 1]
```diff
@@ -25,6 +25,7 @@ def load_audio(
         frames=-1,
         start=0,
         stop=None,
+        channel=None,
```
Comment (boeddeker) [inline paderbox/io/audioread.py:28]: Add to docstring

File: paderbox/io/audioread.py [hunk 2]
```diff
@@ -228,17 +269,83 @@ def load_audio(
...
 
-    # When signal is multichannel, than soundfile return (samples, channels)
+    # When signal is multichannel, then soundfile returns (samples, channels)
     # At NT it is more common to have the shape (channels, samples)
     # => transpose
     signal = signal.T
 
+    # Slice along channel dimension if channel_slice is given
+    if channel is not None:
+        signal = signal[channel,]
```
Comment (boeddeker) [inline paderbox/io/audioread.py:279]: Check that signal has two axis. Soundfile does a squeeze by default

**Comment by boeddeker** [comment]
We should document, that we used pytest as inspiration for the notation

**Comment by thequilo** [comment]
The tests work locally for me (Python 3.10). I have to check what goes wrong on GitHub

File: paderbox/io/audioread.py [hunk 3]
```diff
@@ -228,17 +261,84 @@ def load_audio(
...
 
-    # When signal is multichannel, than soundfile return (samples, channels)
+    # When signal is multichannel, then soundfile returns (samples, channels)
     # At NT it is more common to have the shape (channels, samples)
     # => transpose
     signal = signal.T
 
+    # Slice along channel dimension if channel_slice is given
+    if channel is not None:
+        assert signal.ndim == 2 or channel == 0, (signal.shape, channel)
```
[the remark this answers is not in the document]
Comment (thequilo) [reply paderbox/io/audioread.py:271]: Exactly. I think this shouldn't fail when `channel = 0`  and a file with only a single channel. But I realize now that then also `channel = slice(None)` should pass through

File: paderbox/io/audioread.py [hunk 4]
```diff
@@ -121,6 +126,27 @@ def load_audio(
...
+    >>> data = load_audio(path)
+    >>> data.shape
+    (6, 38520)
+    >>> data = load_audio(str(path) + '::[:,0]')
+    >>> data.shape
+    (38520,)
+    >>> data = load_audio(str(path) + '::[:,:2]')
+    >>> data.shape
+    (2, 38520)
+    >>> data = load_audio(str(path) + '::[:,2:4]')
```
Comment (boeddeker) [inline paderbox/io/audioread.py:144]: Can you add a test with pathlib.Path?

**Review by boeddeker — APPROVED** [review]

**Pull request merged**